packages/server/src/server.js

```diff
@@ -79,10 +79,19 @@ export const handle = async (server, request) => {
     }
   } else {
     const { encoder, decoder } = selection.ok
-    const message = await decoder.decode(request)
-    const result = await execute(message, server)
-    const response = await encoder.encode(result)
-    return response
+    try {
+      const message = await decoder.decode(request)
```

Comment:
We only need the try/catch around the decode, otherwise legitimate handler execution errors will also be returned as a 400.